### PR TITLE
Unit test to include/exclude news sources

### PR DESCRIPTION
**Pull request by Sheena309**
Added a test file: test_filter_sites.py

### CODE REVIEW

File: tests/test_filter_sites.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+@pytest.fixture
+def sample_sites():
+    # Returns actual site configurations loaded from YAML files
+    return sites
+
+
+def test_include_option(sample_sites):
+    # Print the number of YAML files loaded
+    total_sites = len(sample_sites)
+    print(f"Total number of YAML files: {total_sites}")
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:21]: let's remove these print statements

File: tests/test_filter_sites.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        headlines=[],
+        grouped_headlines={},
+        documents_for_display={}
+    )
+    filter_sites = FilterSites()
+    result_context = filter_sites.run(context)
+
+     # Check if only the included sites are present
+    expected_abbreviations = {"nyt", "wsj"}
+    print(f"Expected sites length: {len(expected_abbreviations)}")
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:36]: remove

File: tests/test_filter_sites.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    result_abbreviations = {site.abbreviation for site in result_context.site_configs}
+    assert len(result_context.site_configs) == len(expected_abbreviations)
+    assert result_abbreviations == expected_abbreviations
+    print("***test_include_option passed***")
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:40]: remove

File: tests/test_filter_sites.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    # Check if only the included site is present
+    assert len(result_context.site_configs) == 1
+    assert result_context.site_configs[0].abbreviation == "nyt"
+    assert result_context.site_configs[0].abbreviation == "nyt" 
+    '''
+
+def test_exclude_option(sample_sites):
+    # Print the number of YAML files loaded
+    total_sites = len(sample_sites)
+    print(f"Total number of YAML files: {total_sites}")
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:52]: remove

File: tests/test_filter_sites.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    )
+    filter_sites = FilterSites()
+    result_context = filter_sites.run(context)
+
+    # Check if the excluded site is not present and the count is correct
+    expected_length = total_sites - 1
+    print(f"Expected length of YAML files: {expected_length}")
+    assert len(result_context.site_configs) == expected_length
+    assert all(site.abbreviation != "nyt" for site in result_context.site_configs)
+    print("***test_exclude_option passed***")
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:70]: remove

File: tests/test_filter_sites.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        headlines=[],
+        grouped_headlines={},
+        documents_for_display={}
+    )
+    filter_sites = FilterSites()
+    result_context = filter_sites.run(context)
+
+    # Check if the excluded site is not present and the count is correct
+    expected_length = total_sites - 1
+    print(f"Expected length of YAML files: {expected_length}")
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:67]: remove

File: tests/test_filter_sites.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        user_input=user_input,
+        headlines=[],
+        grouped_headlines={},
+        documents_for_display={}
+    )
+    filter_sites = FilterSites(context)
+    result_context = filter_sites.run()
+    assert len(result_context.site_configs) == len(sample_sites)
+    assert result_context.site_configs == sample_sites
+'''
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:102]: remove commented code

File: tests/test_filter_sites.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        user_input=user_input,
+        headlines=[],
+        grouped_headlines={},
+        documents_for_display={}
+    )
+    filter_sites = FilterSites()
+    #result_context = filter_sites.run(context)
+    with pytest.raises(click.BadParameter):
+        filter_sites.run(context)
+    print("***test_include_and_exclude_option passed***")
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:86]: remove

File: tests/test_filter_sites.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    assert len(result_context.site_configs) == len(expected_abbreviations)
+    assert result_abbreviations == expected_abbreviations
+    print("***test_include_option passed***")
+
+    '''
+    # Check if only the included site is present
+    assert len(result_context.site_configs) == 1
+    assert result_context.site_configs[0].abbreviation == "nyt"
+    assert result_context.site_configs[0].abbreviation == "nyt" 
+    '''
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:47]: remove commented code

File: tests/test_filter_sites.py [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+def sample_sites():
+    # Returns actual site configurations loaded from YAML files
+    return sites
+
+
+def test_include_option(sample_sites):
+    # Print the number of YAML files loaded
+    total_sites = len(sample_sites)
+    print(f"Total number of YAML files: {total_sites}")
+    #user_input = UserPreferences(include=["nyt"])
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:22]: remove comment

File: tests/test_filter_sites.py [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+#comamnd to run this test file : pytest -s .\test_filter_sites.py
+
+@pytest.fixture
+def sample_sites():
+    # Returns actual site configurations loaded from YAML files
+    return sites
+
+
+def test_include_option(sample_sites):
+    # Print the number of YAML files loaded
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:19]: remove comment

File: tests/test_filter_sites.py [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+from src.headline_grabber.pipeline_steps.filter_sites import FilterSites
+from src.headline_grabber.configurations.sites import sites
+from src.headline_grabber.models.news_site import NewsSite
+
+#comamnd to run this test file : pytest -s .\test_filter_sites.py
+
+@pytest.fixture
+def sample_sites():
+    # Returns actual site configurations loaded from YAML files
+    return sites
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:15]: use the `copy()` method on this for safety

File: tests/test_filter_sites.py [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+    '''
+    # Check if only the included site is present
+    assert len(result_context.site_configs) == 1
+    assert result_context.site_configs[0].abbreviation == "nyt"
+    assert result_context.site_configs[0].abbreviation == "nyt" 
+    '''
+
+def test_exclude_option(sample_sites):
+    # Print the number of YAML files loaded
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:50]: remove comment

File: tests/test_filter_sites.py [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+    user_input = UserPreferences(include=["nyt"], exclude=["bbc"])
+    context = PipelineContext(
+        site_configs=sample_sites,
+        user_input=user_input,
+        headlines=[],
+        grouped_headlines={},
+        documents_for_display={}
+    )
+    filter_sites = FilterSites()
+    #result_context = filter_sites.run(context)
```
Comment (bdfinlayson) [inline tests/test_filter_sites.py:83]: remove commented code

**Review by bdfinlayson — CHANGES_REQUESTED** [review]
remove all print statements and commented code

File: src/headline_grabber/pipeline_steps/filter_sites.py [hunk 15]
```diff
@@ -1,17 +1,17 @@
...
 from headline_grabber.pipeline_steps.pipeline_step import PipelineStep
 
-
 class FilterSites(PipelineStep):
     def run(self, context: PipelineContext) -> PipelineContext:
         if context.user_input.include and context.user_input.exclude:
             raise click.BadParameter(
                 f"--include and --exclude are mutually exclusive. Use --include or --exclude to specify the sources to include or exclude."
             )
+
```
Comment (bdfinlayson) [inline src/headline_grabber/pipeline_steps/filter_sites.py:14]: remove extra line

**Review by bdfinlayson — CHANGES_REQUESTED** [review]

**Review by bdfinlayson — APPROVED** [review]
looking good thanks @Sheena309

**Comment by tonthattuit** [comment]
@Sheena309

**Review by tonthattuit — APPROVED** [review]
Looks good to me. Approved.

**Pull request merged**